cpp_exercise/tests/linuxipc_test.cc

```diff
@@ -0,0 +1,183 @@
+
+    remove("thisfiledoesnotexist.txt");
+    std::string exception;
+
+     try{
+       nonexFile.readFile(10);
+       
+       }catch(IPCException & e){
+          exception = e.what();
+          ASSERT_EQ("FileHandler ERROR: Cannot Close the File Correctly", exception);
+          
+       }
+
+    
+  }
+
+   TEST_F(FileHandlerTests, GettingTheCorrectSize){
+     samplevec.insert(samplevec.begin(), samplestring.begin(), samplestring.end());
+    try{
+        txtFile.createFile();
+        txtFile.writeFile(samplevec,samplevec.size());
+    } catch(IPCException & e){}
+         
+       ASSERT_EQ(txtFile.getSize(), samplevec.size() );
+
+  }
+
+
+   TEST_F(FileHandlerTests, ReadingAndWritingTheCorrectData){
+     samplevec.insert(samplevec.begin(), samplestring.begin(), samplestring.end());
+     std::vector<unsigned char> testvec;
+    try{
+        txtFile.createFile();
+        txtFile.writeFile(samplevec,samplevec.size());
+        testvec = txtFile.readFile(samplevec.size());
+    } catch(IPCException & e){}
+      remove("testfile.txt"); 
+       ASSERT_EQ(testvec, samplevec );
+
+  }
+
+
+
+   TEST_F(FileHandlerTests, RemovingFile){
+
+    samplevec.insert(samplevec.begin(), samplestring.begin(), samplestring.end());
+    std::string exception;
+    std::vector<unsigned char> testvec;
+    try{
+        txtFile.createFile();
+        txtFile.writeFile(samplevec,samplevec.size());
+        txtFile.removeFile();
+        testvec = txtFile.readFile(samplevec.size());
+    } catch(IPCException & e){
+          exception = e.what();
+          ASSERT_EQ("FileHandler ERROR: Cannot Close the File Correctly", exception);
+    }
+       
+    }
+
+ 
+
+/**
+ * Testing CommandOption
+ * 
+ */
+
+
+
+
+TEST(CommandOptionTests, GettingProtocolandFileName){
+  
+  char * arggv[4] = {"","--pipe","--file","test.txt"};
+ 
+
+  CommandOption co("ipcsender",4,arggv);
+  std::vector<std::string> outputs = co.getCommand();
+  ASSERT_EQ(outputs[0],"test.txt");
+  ASSERT_EQ(outputs[1],"pipe");
+
+}
+
+TEST(CommandOptionTests, GivingOnlyProtocolAsArgument){
+  
+  char * arggv[2] = {"","--pipe"};
+  std::string exception;
+ try{
+    CommandOption co("ipcsender",2,arggv);
+  std::vector<std::string> outputs = co.getCommand();
+
+ } catch(IPCException & e){
+   exception = e.what();
+   
+   ASSERT_EQ("unrecognized command. filename and protocol should be determined. please use \"--help\" for guide.\n",exception);
+ }  
+  
+}
+
+
+TEST(CommandOptionTests, GivingWrongCommand){
+  
+  char * arggv[2] = {"","--wrongcommand"};
+  std::string exception;
+ try{
+    CommandOption co("ipcsender",2,arggv);
+  std::vector<std::string> outputs = co.getCommand();
+
+ } catch(IPCException & e){
+   exception = e.what();
+   ASSERT_EQ("unrecognized command. please use \"--help\" for guide.\n",exception);
+ }  
+  
+}
+
+TEST(CommandOptionTests, GivingOnlyFileAsArgument){
+  
+  char * arggv[3] = {"","--file","test.txt"};
```

Comment:
it is forbidden to convert const char* ("--file") to char*

Reply:
I tried different ways to fix that but the only solution is to change my main functions to accept only const which I think since this kind of convert is only for testing purposes is not going to be that much harmful (?).

Reply:
``char cmdlineTemp[][4096] = {"myProgramName", "--help"};
  char *argv[] = {cmdlineTemp[0], cmdlineTemp[1], NULL};
``

Reply:
Thank u so much. It worked